doc/user/content/overview/arrangements.md

```diff
@@ -0,0 +1,130 @@
+
+When you create a materialized view and issue a query, Materialize creates a **dataflow**. A dataflow consists of instructions on how to respond to data input. Once executed, the dataflow computes and stores the result of the SQL query in memory, polls the source for updates, and then incrementally updates the query results when new data arrives.
+
+### Collections
+
+Materialize dataflows act on **collections** of data, [multisets](https://en.wikipedia.org/wiki/Multiset) that store each event in an update stream as a triple of `(data, time, diff)`.
+
+Term | Definition
+-----|-----------
+**data**  |  The record update.
+**time**  |  The logical timestamp of the update.
+**diff**  |  Number of copies of the record (`-1` for deletion, `1` for addition).
+
+## Arrangements
+
+To transform the source input into the dataflow into the result output, operators (independent computational components)
+build and maintain indexes on the input and output collections so that they can quickly access collection data in the future. Because queries can overlap, different operators often need to build the exact same indexes for multiple queries. Instead of asking operators to perform redundant work, Materialize builds the index once and maintains it in memory, sharing the required resources across all the operators that use the indexed data. The index is then effectively a sunk cost, and the cost of each query is determined only by the new work it introduces.
+
+We call the indexed update stream an **arrangement**.
+
+In addition to materialized views, certain internal operators, like `reduce` and `join`, also create arrangements so the operators can respond efficiently to updates.
+
+### Arrangement structure
+
+In the `(data, time, diff)` triples that make up the update stream, `data` is structured as a key-value pair.  Arrangements are indexed on the `data` keys. You cannot manually create an arrangement, although you can influence the shape of the arrangements created by Materialize by the way you structure queries and views.
+
+When creating an arrangement for a join where the user has not specified a key, Materialize attempts to choose a key that will ensure that data is well distributed. If there is a primary key, that will be used; if there are source fields not required by the query, they are not included. Often Materialize can pull primary key info from a Confluent schema.
+
+### Arrangement size
+
+For joins, the size of an arrangement, or amount of memory it requires, is roughly equivalent to its number of distinct `(data, time)` pairs, which can be small even if the number of records is large. As an illustration, consider a histogram of taxi rides grouped by the number of riders and the fare amount. The number of `(rider, fare)` pairs will be much smaller than the number of total rides that take place.
```

Comment:
I don't think it has anything to do with joins. Independent of joins or not, all arrangements take size proportional to the number of distinct `(data, time)` pairs in the arranged data. I might put "distinct" in "The number <'distinct' here> of `(rider, ...)`"

Reply:
I would say "proportional" rather than "equivalent".